src/core/annotation.js

```diff
@@ -2483,6 +2494,133 @@ class ChoiceWidgetAnnotation extends WidgetAnnotation {
       type,
     };
   }
+
+  async _getAppearance(evaluator, task, annotationStorage) {
+    if (this.data.combo) {
+      return super._getAppearance(evaluator, task, annotationStorage);
+    }
+
+    if (!annotationStorage) {
+      return null;
+    }
+    const storageEntry = annotationStorage.get(this.data.id);
+    let exportedValue = storageEntry && storageEntry.value;
+    if (exportedValue === undefined) {
+      // The annotation hasn't been rendered so use the appearance
+      return null;
+    }
+
+    if (!Array.isArray(exportedValue)) {
+      exportedValue = [exportedValue];
+    }
+
+    const defaultPadding = 2;
+    const hPadding = defaultPadding;
+    const totalHeight = this.data.rect[3] - this.data.rect[1];
+    const totalWidth = this.data.rect[2] - this.data.rect[0];
+    const lineCount = this.data.options.length;
+    const valueIndice = [];
+    for (let i = 0, ii = this.data.options.length; i < ii; i++) {
```

Comment:
Nit: How about just using `i < lineCount` here, since the length was already determined just above?